test/functionalTest/cases/4541_wrong_date_values_in_subscription/wrong_date_values_in_expires_field.test

```diff
@@ -0,0 +1,358 @@
+  },
+  "notification": {
+    "http": {
+      "url": "http://localhost:'$LISTENER_PORT'/notify"
+    },
+    "attrs": [
+      "pressure"
+    ]
+  },
+  "expires": "2022-12-21T17:65:03.00Z"
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "03. Create a subscription with an wrong date value in the 'hours' component of the 'expires' field, see error"
+echo "============================================================================================================="
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "id": "Room1",
+        "type": "Room"
+      }
+    ],
+    "condition": {
+      "attrs": [
+        "pressure"
+      ]
+    }
+  },
+  "notification": {
+    "http": {
+      "url": "http://localhost:'$LISTENER_PORT'/notify"
+    },
+    "attrs": [
+      "pressure"
+    ]
+  },
+  "expires": "2022-12-21T25:16:11.00Z"
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "04. Create a subscription with an wrong date value in the 'days' component of the 'expires' field, see error"
+echo "============================================================================================================"
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "id": "Room1",
+        "type": "Room"
+      }
+    ],
+    "condition": {
+      "attrs": [
+        "pressure"
+      ]
+    }
+  },
+  "notification": {
+    "http": {
+      "url": "http://localhost:'$LISTENER_PORT'/notify"
+    },
+    "attrs": [
+      "pressure"
+    ]
+  },
+  "expires": "2022-12-34T17:16:23.00Z"
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "05. Create a subscription with an wrong date value in the 'months' component of the 'expires' field, see error"
+echo "=============================================================================================================="
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "id": "Room1",
+        "type": "Room"
+      }
+    ],
+    "condition": {
+      "attrs": [
+        "pressure"
+      ]
+    }
+  },
+  "notification": {
+    "http": {
+      "url": "http://localhost:'$LISTENER_PORT'/notify"
+    },
+    "attrs": [
+      "pressure"
+    ]
+  },
+  "expires": "2022-13-21T17:16:43.00Z"
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "06. Create a subscription with an wrong date value in the 'years' component of the 'expires' field, see error"
+echo "============================================================================================================="
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "id": "Room1",
+        "type": "Room"
+      }
+    ],
+    "condition": {
+      "attrs": [
+        "pressure"
+      ]
+    }
+  },
+  "notification": {
+    "http": {
+      "url": "http://localhost:'$LISTENER_PORT'/notify"
+    },
+    "attrs": [
+      "pressure"
+    ]
+  },
+  "expires": "1822-12-21T17:16:23.00Z"
```

Comment:
This is a year in XIX century :), but from a point of view of the checking logic 1822 is greater than `minYear = 0;`.

So why is this an invalid year? Maybe I'm missing something...

Reply:
In this test case, I was checking valid year. As per existing code in `globals.cpp` file
```
  struct tm time;
  time.tm_year = y - 1900; // Year since 1900
  time.tm_mon  = M - 1;    // 0-11
  time.tm_mday = d;        // 1-31
  time.tm_hour = h;        // 0-23
  time.tm_min  = m;        // 0-59
  time.tm_sec  = (int) s;  // 0-61 (0-60 in C++11)
```
In above code, The base year has been set to 1900. Therefore, any year earlier than 1900 should be considered invalid. In that context, I have randomly chosen the year 1822 as an invalid date in the test case.

Reply:
> but from a point of view of the checking logic 1822 is greater than minYear = 0;.
In the existing code, the original year value is subtracted by 1900.
so 1822 or 1899 is less than 0. (1899-1900)

Reply:
Ok, I didn't remember that substraction :) Makes sense

Maybe this should be warned in the new code (I provide a suggestion). With regards to this comment thread, NTC

Reply:
> Maybe this should be warned in the new code (I provide a suggestion)

Not really needed... looking the code in context, it's clear enough as it is now.